subworkflows/local/compartments_caller/hicexplorer.nf

```
@@ -0,0 +1,49 @@
+/*
+ * call TADs by HiCExploer
+ */
+
+include { HICEXPLORER_HICPCA          } from '../../../modules/local/hicexplorer/hicpca'
+include { HICEXPLORER_HICTRANSFORM    } from '../../../modules/local/hicexplorer/hictransform'
+include { HICEXPLORER_HICPLOTMATRIX
+    as HICEXPLORER_HICPLOTMATRIX_PCA1 } from '../../../modules/local/hicexplorer/hicplotmatrix'
+include { HICEXPLORER_HICPLOTMATRIX
+    as HICEXPLORER_HICPLOTMATRIX_PCA2 } from '../../../modules/local/hicexplorer/hicplotmatrix'
+
+workflow HICEXPLORER_COMPARTMENTS {
+    take:
+    cool       // channel: [ val(meta), [cool] ]
+    resolution // channel: [ val(resolution) ]
+    chromsizes // channel: [ path(size) ]
+
+    main:
+    ch_multiqc_files = Channel.empty()
+    ch_versions      = Channel.empty()
+
+    // step1 create bigwig files, the input file is balanced by cooler::balance (aka, ICE)
+    HICEXPLORER_HICPCA(
+        cool
+    )
+    ch_version = HICEXPLORER_HICPCA.out.versions.ifEmpty(null)
+
+    // step2 create intermeidated files for plot
+    HICEXPLORER_HICTRANSFORM(
+        cool
+    )
+    ch_version = ch_version.mix(HICEXPLORER_HICTRANSFORM.out.versions.ifEmpty(null))
```

Are you sure you want the ifEmpty everywhere? You are definitely executing the module - if it doesn't produce a version this indicates a tool error for some reason... here you could be hiding such an error by saying if empty `[]`

---

`ifEmpty(null)` will be removed.